### PR TITLE
Add SSHKDF in evp_kdf_test

### PR DESCRIPTION
**Pull request by simo5**
##### Checklist
- [X] tests are added or updated

### CODE REVIEW

File: test/evp_kdf_test.c [hunk 1]
```diff
@@ -253,6 +253,57 @@ static int test_kdf_ss_kmac(void)
...
+        0xb2, 0x74, 0x5e, 0xbc, 0x99, 0xb2, 0x67, 0xda, 0xa8, 0x6a, 0x4a, 0xa7,
+        0x69, 0x72, 0x39, 0x08, 0x82, 0x53, 0xf6, 0x42
+    };
+    const unsigned char sessid[] = {
+        0x0e, 0x68, 0x3f, 0xc8, 0xa9, 0xed, 0x7c, 0x2f, 0xf0, 0x2d, 0xef, 0x23,
+        0xb2, 0x74, 0x5e, 0xbc, 0x99, 0xb2, 0x67, 0xda, 0xa8, 0x6a, 0x4a, 0xa7,
+        0x69, 0x72, 0x39, 0x08, 0x82, 0x53, 0xf6, 0x42
+    };
+    const unsigned char expected[sizeof(out)] = {
+        0x41, 0xff, 0x2e, 0xad, 0x16, 0x83, 0xf1, 0xe6
```
Comment (mattcaswell) [inline test/evp_kdf_test.c]: Is this a self generated test vector?

Comment (simo5) [reply test/evp_kdf_test.c]: No it is the first sha256 test we have already in evpkdf.txt which is from CAVS vectors

Comment (mattcaswell) [reply test/evp_kdf_test.c]: Ah, ok. A comment stating its original source would be good.

**Comment by simo5** [comment]
Fixed the type parameter

File: test/evp_kdf_test.c (hunk 1 above)
Comment (simo5) [reply test/evp_kdf_test.c]: No other tests data in that file is commented, but I can add a comment.

**Comment by simo5** [comment]
And added comment on test data

**Comment by slontis** [comment]
The other bit I was talking about is the ctrl string name for the digest in the SSHKDF code.
All other KDF's are using the name "digest" instead of "md". The EVP_KDF_CTX.pod documentation is incorrectly saying that it uses "md" (this has been updated in the other PR).

**Comment by simo5** [comment]
> The other bit I was talking about is the ctrl string name for the digest in the SSHKDF code.
> All other KDF's are using the name "digest" instead of "md". The EVP_KDF_CTX.pod documentation is incorrectly saying that it uses "md" (this doc been updated in PR #8762).

Uhm this is sort of a breaking change (patches have already been backpored), ok If I change to advertise digest by default in docs, but still support the "md" string as well as an alias ?

**Comment by simo5** [comment]
I'll open a sparate PR for the digest change, this one is good to go IMO.

**Comment by mattcaswell** [comment]
I've pushed it.

**Pull request closed without merging**